extensions/amp-document-recommendations/0.1/amp-document-recommendations.js

```diff
@@ -229,6 +227,42 @@ export class AmpDocumentRecommendations extends AMP.BaseElement {
     this.appendNextArticle_();
     return Promise.resolve();
   }
+
+  /**
+   * Called from a {@link AmpDocumentRecommendationsMarker} element's
+   * layoutCallback, indicating the next article should be prerendered.
+   */
+  markerLayoutCallback() {
+    if (this.nextArticle_ < this.config_.recommendations.length) {
+      this.appendNextArticle_();
+    }
+  }
 }
 
 AMP.registerElement(TAG, AmpDocumentRecommendations, CSS);
+
+/** @const */
+const MARKER_TAG = 'amp-document-recommendations-marker';
+
+class AmpDocumentRecommendationsMarker extends AMP.BaseElement {
+
+  /** @override */
+  isLayoutSupported(unused) {
+    return true;
+  }
+
+  /** @override */
+  buildCallback() {
+    // TODO(peterjosling): reuse this as the fallback button
```

Comment:
What is the fallback button?

Reply:
The "click to go to article" button we talked about having if it failed to load in time. But we're not having that button anymore, are we...